src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -95,6 +97,7 @@ public CRAMFileReader(final File cramFile, final InputStream inputStream,
         this.cramFile = cramFile;
         this.inputStream = inputStream;
         this.referenceSource = referenceSource;
+        if (cramFile != null) initIndexFile(null);
```

Comment:
Since initIndexFile is called in the middle of the constructors, I would make it static (you'll have to pass in cramFile as well) and just have it return the updated indexFile, which the constructors can use to initialize the member variable. You might rename it (maybe "findIndexForFile" ?).